lib/Files/TemplateLoader.php

```diff
@@ -23,34 +23,41 @@
 
 namespace OCA\Talk\Files;
 
+use OCA\Files\Event\LoadSidebar;
+use OCA\Talk\AppInfo\Application;
+use OCP\EventDispatcher\Event;
 use OCP\EventDispatcher\IEventDispatcher;
+use OCP\EventDispatcher\IEventListener;
 use OCP\Util;
 
 /**
  * Helper class to add the Talk UI to the sidebar of the Files app.
  */
-class TemplateLoader {
+class TemplateLoader implements IEventListener {
 
 	public static function register(IEventDispatcher $dispatcher): void {
-		$dispatcher->addListener('OCA\Files::loadAdditionalScripts', static function() {
-			self::loadTalkSidebarForFilesApp();
-		});
+		$dispatcher->addServiceListener(LoadSidebar::class, TemplateLoader::class);
 	}
 
 	/**
 	 * Loads the Talk UI in the sidebar of the Files app.
 	 *
-	 * This method should be called when loading additional scripts for the
+	 * This method should be called when handling the LoadSidebar event of the
 	 * Files app.
 	 */
-	public static function loadTalkSidebarForFilesApp(): void {
+	public function handle(Event $event): void {
+		if (!($event instanceof LoadSidebar)) {
```

Comment:
I would argue that this is far less cleaner and different that what we now encourage people to do on other apps ^^

Reply:
But we can still merge it, there is bigger priorities ;)

Reply:
What is "_far less cleaner and different that what we now encourage people to do on other apps_"? Checking if the event is an instance of `LoadSidebar`? Because I just copied it over from [your code](https://github.com/nextcloud/spreed/pull/2549/files#diff-4548f384ebc3e4192399f4922a6c1612R37) :-P

If it is using a class named `TemplateLoader` instead of `LoadSidebarListener` I do not think it is so bad, and it is consistent with how the other sidebars (public share and public share authentication) are loaded in Talk.

And if it is a different thing, could you provide more context about what is wrong so it can be fixed later? Thanks :-)

Reply:
> If it is using a class named `TemplateLoader` instead of `LoadSidebarListener` I do not think it is so bad, and it is consistent with how the other sidebars (public share and public share authentication) are loaded in Talk.

just this part yes, I find it easier to search for the listener when I know the event is being listened to than searching for a file called "TemplateLoader" :woman_shrugging: 
But that is nitpicking! Let's get this in!